Automated Smoketest: :x:
OS: centos 7.4.1708 (Linux 3.10.0-327.28.3.el7.x86_64)
Sha: e6bc6506627beb277e41783cf48e19275178735d
Number of Error(s): 4
-- Configuring incomplete, errors occurred!
Number of Error(s): 4
-- Configuring incomplete, errors occurred!
make[2]: *** [plugins/couchbase/libcouchbase-build/lib/libcouchbase.so] Error 1
make[1]: *** [plugins/couchbase/CMakeFiles/libcouchbase-build.dir/all] Error 2
make: *** [all] Error 2
CMakeFiles/CMakeOutput.log copied into ../CMakeOutput-2019-03-11_13-47-47.log
CMakeFiles/CMakeError.log copied into ../CMakeError-2019-03-11_13-47-47.log

[Build error:](https://gist.github.com/c133bb3ae6c8c12bb69ee1cfbe6d6142)
Build log [Build log](https://gist.github.com/HPCCSmoketest/c133bb3ae6c8c12bb69ee1cfbe6d6142#file-result-19-03-11-13-47-27-log)

--- a/ecl/hql/hqlgram2.cpp
+++ b/ecl/hql/hqlgram2.cpp
@@ -918,6 +918,12 @@ IHqlExpression * HqlGram::processEmbedBody(const attribute & errpos, IHqlExpress
     if (!type)
         type.setown(makeVoidType());
 
+    if (type->getTypeCode()!=type_table && type->getTypeCode()!=type_groupedtable && type->getTypeCode()!=type_void)
+    {
+        if (attribs && queryAttributeInList(activityAtom,attribs))
+             reportError(ERR_EMBEDERROR, errpos, "Embedded activity only supports dataset return type or action");
+    }
+
     if (type->getTypeCode() == type_record)
         type.setown(makeRowType(LINK(type)));
 

--- a/ecl/regress/embedactivity2err.ecl
+++ b/ecl/regress/embedactivity2err.ecl
@@ -1,0 +1,8 @@
+rec := { string f1; };
+
+UNSIGNED4 myfunc(STREAMED DATASET(rec) recs) := EMBED(C++: activity)
+  #body
+  return 0 
+ENDEMBED;
+
+myfunc(DATASET([{'f1'}], rec));

